src/v/cluster/tx_utils.h

```diff
@@ -0,0 +1,20 @@
+#include "cluster/types.h"
+#include "config/configuration.h"
+#include "kafka/protocol/errors.h"
+#include "model/fundamental.h"
+#include "model/record.h"
+#include "raft/consensus.h"
+#include "raft/errc.h"
+#include "raft/logger.h"
+#include "raft/state_machine.h"
+#include "raft/types.h"
+#include "storage/snapshot.h"
+#include "utils/expiring_promise.h"
+#include "utils/mutex.h"
+
+namespace cluster {
+
+model::record_batch
```

Comment:
nit: we can probably forward declare the types see any of the `fwd.h` files for example.

Reply:
tx_util introduces only a free function - `model::record_batch make_fence_batch(model::control_record_version, model::producer_identity);` while `fwd.h` defines only classes do you want `make_fence_batch` to be the first function there?

Reply:
makes sense.